c7n/offhours.py

```diff
@@ -277,10 +294,41 @@ def get_local_tz(self, parts):
             self.log.warning(
                 "filter:offhours unknown tz %s for %s" % (
                     tz_spec, parts))
-                    
+
             return None
         return tz
-    
+
+
+    def is_custom(self, parts):
+        if len(parts) == 3 and parts.get("on") and parts.get("off"):
+            return True
+        else:
+            return False
+
+    def get_custom_time(self, i, now, parts):
+        if type(self).__name__ in ("InstanceOnHour", "OnHour"):
```

Comment:
the later would be better, ie. attribute match on class. allows for flexibility of subclass without hardcoded strings in the parent. the former mention was just educational in nature. Thanks! jealous re pycon ;-)